content/blog/2017/2017-01-18-refactoring-conditional-to-pipeline.adoc

```diff
@@ -0,0 +1,475 @@
+            description: '',
+            name : 'BRANCH_PATTERN')
+        booleanParam (
+            defaultValue: false,
+            description: '',
+            name : 'FORCE_FULL_BUILD')
+    }
+
+    stages {
+        stage ('Prepare') {
+            steps {
+                checkout([$class: 'GitSCM',
+                    branches: [[name: "origin/${BRANCH_PATTERN}"]],
+                    doGenerateSubmoduleConfigurations: false,
+                    extensions: [[$class: 'LocalBranch']],
+                    submoduleCfg: [],
+                    userRemoteConfigs: [[credentialsId: '9f926524-9fac-44df-99dd-53069380f286', url: 'https://github.com/bitwiseman/hermann']]])
+            }
+        }
+
+        stage ('Build') {
+            when {
+                expression {
+                    GIT_BRANCH = 'origin/' + sh(returnStdout: true, script: 'git rev-parse --abbrev-ref HEAD').trim()
+                    return GIT_BRANCH == 'origin/master' || params.FORCE_FULL_BUILD
+                }
+            }
+            steps {
+                parallel (
+                    linux: {
+                        build job: 'full-build-linux', parameters: [string(name: 'GIT_BRANCH_NAME', value: GIT_BRANCH)]
+                    },
+                    mac: {
+                        build job: 'full-build-mac', parameters: [string(name: 'GIT_BRANCH_NAME', value: GIT_BRANCH)]
+                    },
+                    windows: {
+                        build job: 'full-build-windows', parameters: [string(name: 'GIT_BRANCH_NAME', value: GIT_BRANCH)]
+                    },
+                    failFast: false)
+            }
+        }
+        stage ('Build Skipped') {
+            when {
+                expression {
+                    GIT_BRANCH = 'origin/' + sh(returnStdout: true, script: 'git rev-parse --abbrev-ref HEAD').trim()
+                    return !(GIT_BRANCH == 'origin/master' || params.FORCE_FULL_BUILD)
+                }
+            }
+            steps {
+                echo 'Skipped full build.'
+            }
+        }
+    }
+}
+----
+
+
+== Conclusion
+
+As I said before, the Conditional BuildStep plugin is great.
+It provides a clear, easy to understand way to add conditional logic to any freestyle job.
+Before Pipeline, it was one of the few plugins to do this and it remains one of the most popular plugins.
+But now there's Jenkins Pipeline.
+And with the upcoming releases of Declarative Pipeline and Blue Ocean, it is time to
+begin looking at migrating from
+
+This is blog post discussed how to approach migrating conditional build steps to pipeline
+and showed a couple concrete examples.  Overall, I'm pleased with the results so far.
+I found scenarios which could not easily be migrated to Pipeline, but even those
+are only more difficult not impossible.
+
+The next thing to do is add a section to the
+link:https://jenkins.io/doc/book/[Jenkins Handbook] documenting the Pipeline
```

Comment:
Very weird phrasing. Why not "We don't have it yet, but are planning to add this soon." or similar?